### PR TITLE
Be/feature/form-fields-enhancement

### PR DESCRIPTION
**Pull request by rburaksaritas**
- Added Recurrence fields to Resource and Need:
  - recurrence_id
  - recurrence_rate
  - recurrence_deadline
  - occur_at
- Updated subtype and type structure.
- Removed unused form fields from form-fields.json and endpoints:
```
GET /api/form_fields/need
{
    "fields": [
        {
            "name": "description",
            "label": "Description",
            "type": "text"
        },
        {
            "name": "initialQuantity",
            "label": "Initial Quantity",
            "type": "number"
        },
        {
            "name": "urgency",
            "label": "Urgency",
            "type": "select",
            "options": [
                1,
                2,
                3,
                4,
                5
            ]
        },
        {
            "name": "type",
            "label": "Type",
            "type": "select",
            "options": [
                "cloth",
                "food",
                "shelter",
                "medication",
                "transportation",
                "tool",
                "human",
                "other"
            ]
        },
        {
            "name": "x",
            "label": "X Coordinate",
            "type": "number"
        },
        {
            "name": "y",
            "label": "Y Coordinate",
            "type": "number"
        },
        {
            "name": "occur_at",
            "label": "Occur At",
            "type": "date"
        },
        {
            "name": "recurrence_rate",
            "label": "Recurrence Rate",
            "type": "select",
            "options": [
                "none",
                "daily",
                "every 2 days",
                "every 3 days",
                "weekly"
            ]
        },
        {
            "name": "recurrence_deadline",
            "label": "Recurrence Deadline",
            "type": "date"
        }
    ]
}
```
- Added predefined subtype selection options:
```
GET /api/form_fields/type/cloth
{
    "fields": [
        {
            "name": "size",
            "label": "Size",
            "type": "text"
        },
        {
            "name": "gender",
            "label": "Gender",
            "type": "text"
        },
        {
            "name": "age group",
            "label": "Age",
            "type": "select",
            "options": [
                "baby",
                "child",
                "teen",
                "adult",
                "elder",
                "mixed"
            ]
        },
        {
            "name": "subtype",
            "label": "Subtype",
            "type": "select",
            "options": [
                "pants",
                "shirt",
                "t-shirt",
                "coat",
                "boots",
                "shoes"
            ]
        }
    ]
}
``

### CODE REVIEW

File: Disaster-Response-Platform/backend/Controllers/form_fields.json [hunk 1]
```diff
@@ -1,74 +1,80 @@
...
         "fields": [
-            {"name": "_id", "label": "ID", "type": "text"},
-            {"name": "created_by", "label": "Created By", "type": "text"},
+            {"name": "description", "label": "Description", "type": "text"},
             {"name": "initialQuantity", "label": "Initial Quantity", "type": "number"},
-            {"name": "urgency", "label": "Urgency", "type": "number"},
-            {"name": "unsuppliedQuantity", "label": "Unsupplied Quantity", "type": "number"},
-            {"name": "type", "label": "Type", "type": "text"},
+            {"name": "urgency", "label": "Urgency", "type": "select", "options": [1, 2, 3, 4, 5]},
+            {"name": "type", "label": "Type", "type": "select", "options": ["cloth", "food", "shelter", "medication", "transportation", "tool", "human", "other"]},
```
Comment (bgmrsln) [inline Disaster-Response-Platform/backend/Controllers/form_fields.json:20]: can you add drink

File: Disaster-Response-Platform/backend/Controllers/form_fields.json [hunk 2]
```diff
@@ -1,74 +1,80 @@
...
             {"name": "x", "label": "X Coordinate", "type": "number"},
-            {"name": "y", "label": "Y Coordinate", "type": "number"}
+            {"name": "y", "label": "Y Coordinate", "type": "number"},
+            {"name": "occur_at", "label": "Occur At", "type": "date"},
+            {"name": "recurrence_rate", "label": "Recurrence Rate", "type": "select", "options": ["none", "daily", "every 2 days", "every 3 days", "weekly"]},
```
Comment (bgmrsln) [inline Disaster-Response-Platform/backend/Controllers/form_fields.json:24]: maybe we dont need every 2 days and every3 days, can you provide a scenerio for it

**Review by bgmrsln — CHANGES_REQUESTED** [review]

File: Disaster-Response-Platform/backend/Controllers/form_fields.json (hunk 2 above)
Comment (rburaksaritas) [reply Disaster-Response-Platform/backend/Controllers/form_fields.json:24]: > maybe we dont need every 2 days and every3 days, can you provide a scenerio for it

I might want baby diapers for every 3 days, no?

Comment (bgmrsln) [reply Disaster-Response-Platform/backend/Controllers/form_fields.json:24]: I think it can be provided accumulative, so no

**Review by bgmrsln — APPROVED** [review]
Thanks for the changes, LGTM

**Pull request merged**